From 0fac823c0e32cc74af40767f67e1e7eeffa35e0b Mon Sep 17 00:00:00 2001
From: Wenping Zhang <wenping.zhang@rock-chips.com>
Date: Wed, 20 Jun 2018 14:56:37 +0800
Subject: [PATCH] power: bq25700: fix crash issue duaring probe bq25700 driver.

In previous code, driver will update the power supply status before
power_supply_register, so it will cause kernel panic. So we put the
bq25700_power_supply_init before bq25700_init_usb to fix this issue.

Change-Id: Id9d206dbecfdadbd968b5c9d24f85f66ef321b9d
Signed-off-by: Wenping Zhang <wenping.zhang@rock-chips.com>
---
 drivers/power/bq25700_charger.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/power/bq25700_charger.c b/drivers/power/bq25700_charger.c
index d0efab7ce4f4..48f87becf0b6 100644
--- a/drivers/power/bq25700_charger.c
+++ b/drivers/power/bq25700_charger.c
@@ -2015,9 +2015,10 @@ static int bq25700_probe(struct i2c_client *client,
 	}
 
 	bq25700_parse_dt(charger);
+	bq25700_init_sysfs(charger);
 
+	bq25700_power_supply_init(charger);
 	bq25700_init_usb(charger);
-	bq25700_init_sysfs(charger);
 
 	if (client->irq < 0) {
 		dev_err(dev, "No irq resource found.\n");
@@ -2039,7 +2040,6 @@ static int bq25700_probe(struct i2c_client *client,
 		goto irq_fail;
 	enable_irq_wake(client->irq);
 
-	bq25700_power_supply_init(charger);
 	bq25700_charger = charger;
 
 irq_fail:
-- 
2.35.3

